### PR TITLE
Viewports!

### LINKED ISSUE #164: BoundingBox
Currently, boundingbox is only used by tests, even though our game only uses CollisionMasks. We should make it easier for tests to use CollisionMasks to perform the same tests, and remove BoundingBox  completely.

See #162

### PR DESCRIPTION
**Pull request by ChielBruin**
Added viewports to the renderer.
This viewport will follow the player when he moves.

bdf0ed9 and c119847 for the implementation of the viewports, 97c6559 and bee3f42 for some tweaks and bcfc905, 71a4f8b for some bugfixes.

Fixes #164, #191

### CODE REVIEW

**Comment by ChielBruin** [comment]
@jessearens I fixed it!
Thanks for noticing.

<logic, bug

File: src/main/java/com/github/fishio/SinglePlayerPlayingField.java [hunk 1]
```diff
@@ -29,14 +26,18 @@
...
 	 * @param canvas
-	 *            the canvas to use, can be <code>null</code> to create one.
+	 *      the canvas to use, can be <code>null</code> to create one.
 	 * @param scene
-	 *            the scene to use (for registering keylisteners)
+	 *      the scene to use (for registering keylisteners)
+	 * @param width
+	 * 		the width of the playing field
+	 * @param height
+	 * 		the height of the playing field
```
Comment (jessearens) [inline src/main/java/com/github/fishio/SinglePlayerPlayingField.java:37]: Good work on improving the layout of this javadocs. However, I think you forgot the identation of width and height?

<documentation

**Comment by jessearens** [comment]
With my few comments fixed, I approve of this awesome new feature.

<feature

**Comment by Taeir** [comment]
I also ran the game, and it looks so much better with the moving screen!

The code looks good, so this can be merged.

<feature

**Pull request merged**